src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitString.Fixer.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public class PreferConstCharOverConstUnitStringFixer : CodeFixProvider
+    {
+        private const string title = "Replace const unit string with const char";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(PreferConstCharOverConstUnitStringAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            ImmutableArray<Diagnostic> diagnostics = context.Diagnostics;
+            CodeAnalysis.Text.TextSpan diagnosticsSpan = diagnostics.First().Location.SourceSpan;
+
+            // Find the type declaration identified by the diagnostic. Fix only local declarations.
+            IEnumerable<LocalDeclarationStatementSyntax> declarations = root.FindToken(diagnosticsSpan.Start).Parent.AncestorsAndSelf().OfType<LocalDeclarationStatementSyntax>();
```

Comment:
1 decision to be made here is if we want to apply the code fix for class fields that are defined as `const string` too(the analyzer already fires). For ex: Look at the `TestClassField` unit test here: https://github.com/dotnet/roslyn-analyzers/pull/3481/files#diff-b1b9befd17d0df8e9eb932ce2b80ea7eR93

I ran into this case in the runtime repo where the automatic code-fix would lead to compile errors for most cases: For ex: The class field (defined as `const string`) could be passed to a method that only accepts a `string`. The code fix would break all such instances.